ament_package/package.py

```diff
@@ -98,9 +112,13 @@ def validate(self):
 
         if not self.name:
             errors.append('Package name must not be empty')
-        # only allow lower case alphanummeric characters and underscores
-        # must start with an alphabetic character
-        if not re.match('^[a-z][a-z0-9_]*$', self.name):
+        valid_package_name_regexp = '^[a-z][a-z0-9_]*$'
+        # must start with an alphabetic character only allow lower case
+        # alphanummeric characters and underscores in catkin or ament packages.
+        # Dashes are allowed for other build_types.
```

Comment:
This comment line should go before line 120 which contains the regex allowing the dash.